x/ccv/consumer/keeper/relay.go

```diff
@@ -223,6 +236,40 @@ func (k Keeper) SendPackets(ctx sdk.Context) {
 // in conjunction with the ibc module's execution of "acknowledgePacket",
 // according to https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics#processing-acknowledgements
 func (k Keeper) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Packet, ack channeltypes.Acknowledgement) error {
+	if res := ack.GetResult(); res != nil {
+		if len(res) != 1 {
+			k.Logger(ctx).Error("recv invalid ack; expected length 1", "channel", packet.SourceChannel, "ack", res)
+		}
+
+		// Unmarshal into V1 consumer packet data type. We trust data is formed correctly
+		// as it was originally marshalled by this module, and consumers must trust the provider
+		// did not tamper with the data. Note ConsumerPacketData.GetBytes() always JSON marshals to the
+		// ConsumerPacketDataV1 type which is sent over the wire.
+		var consumerPacket ccv.ConsumerPacketDataV1
+		ccv.ModuleCdc.MustUnmarshalJSON(packet.GetData(), &consumerPacket)
+		// If this ack is regarding a provider handling a vsc matured packet, there's nothing to do.
+		// As vsc matured packets are popped from the consumer pending packets queue on send.
+		if consumerPacket.Type == ccv.VscMaturedPacket {
+			return nil
+		}
+
+		// Otherwise we handle the result of the slash packet acknowledgement.
+		switch res[0] {
+		// We treat a v1 result as the provider successfully queuing the slash packet w/o need for retry.
+		case ccv.V1Result[0]:
+			k.ClearSlashRecord(ctx)           // Clears slash record state, unblocks sending of pending packets.
+			k.DeleteHeadOfPendingPackets(ctx) // Remove slash from head of queue. It's been handled.
+		case ccv.SlashPacketHandledResult[0]:
+			k.ClearSlashRecord(ctx)           // Clears slash record state, unblocks sending of pending packets.
+			k.DeleteHeadOfPendingPackets(ctx) // Remove slash from head of queue. It's been handled.
+		case ccv.SlashPacketBouncedResult[0]:
+			k.UpdateSlashRecordOnBounce(ctx)
```

Comment:
Just to verify I understand this correctly: If the provider chain considers itself getting overwhelmed with `SlashPacket`s it would acknowledge pack with a _bounce_ acknowledgment?

Reply:
Yep thats correct 👍 

More info [here](https://cosmos.github.io/interchain-security/adrs/adr-002-throttle) and [here](https://cosmos.github.io/interchain-security/adrs/adr-008-throttle-retries)